vars/checkResultsPublish.groovy

```diff
@@ -0,0 +1,147 @@
+import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.Utils
+
+/**
+ * checkResultsPublish
+ *
+ * @param others document all parameters
+ */
+def call(Map parameters = [:]) {
+    handlePipelineStepErrors (stepName: 'checkResultsPublish', stepParameters: parameters) {
+        // GENERAL
+        def tasks = parameters.get('tasks', false)
+        def aggregation = parameters.get('aggregation', [:])
+        def doArchive = parameters.get('archive', false)
+        // JAVA
+        def pmd = parameters.get('pmd', false)
+        def cpd = parameters.get('cpd', false)
+        def findbugs = parameters.get('findbugs', false)
+        def checkstyle = parameters.get('checkstyle', false)
+        // JAVA SCRIPT
+        def eslint = parameters.get('eslint', false)
+        // PYTHON
+        def pylint = parameters.get('pylint', false)
+
+        // report TODOs
+        reportTasks(tasks, '**/*.java')
```

Comment:
defaults should go into the default configuration yml and merged in through the config framework.